tests/integration/test_models/test_full_battery_models/test_lead_acid/test_full.py

```diff
@@ -1,68 +1,68 @@
-#
-# Tests for the lead-acid Full model
-#
 import pybamm
 import tests
-
+import pytest
 import numpy as np
 
 
 class TestLeadAcidFull:
-    def test_basic_processing(self):
-        options = {"thermal": "isothermal"}
+    @pytest.mark.parametrize(
+        "options,t_eval",
+        [
+            ({"thermal": "isothermal"}, np.linspace(0, 3600 * 17)),
+            (
+                {"thermal": "isothermal", "convection": "uniform transverse"},
+                np.linspace(0, 3600 * 10),
+            ),
+        ],
+        ids=["basic_processing", "basic_processing_with_convection"],
+    )
+    def test_basic_processing(self, options, t_eval):
         model = pybamm.lead_acid.Full(options)
         modeltest = tests.StandardModelTest(model)
-        modeltest.test_all(t_eval=np.linspace(0, 3600 * 17))
+        modeltest.test_all(t_eval=t_eval)
 
-    def test_basic_processing_with_convection(self):
-        options = {"thermal": "isothermal", "convection": "uniform transverse"}
-        model = pybamm.lead_acid.Full(options)
-        # var_pts = {"x_n": 10, "x_s": 10, "x_p": 10}
-        modeltest = tests.StandardModelTest(model)
-        modeltest.test_all(t_eval=np.linspace(0, 3600 * 10))
-
-    def test_optimisations(self):
+    @pytest.fixture
+    def optimisations_test(self):
```

Comment:
Can we move this out of the class + rename it to `optimtest` to maintain consistency?